lua/astrocommunity/remote-development/remote-sshfs/init.lua

```diff
@@ -0,0 +1,47 @@
+---@type LazySpec
+return {
+	"nosduco/remote-sshfs.nvim",
+	config = function(_, opts)
+		require("remote-sshfs").setup(opts)
+		require("telescope").load_extension("remote-sshfs")
+	end,
+	dependencies = {
+		"nvim-telescope/telescope.nvim",
+		"nvim-lua/plenary.nvim",
+	},
+	specs = {
+		"AstroNvim/astrocore",
+		optional = true,
+		opts = function(_, opts)
+			local map = opts.mappings
+			local api = require("remote-sshfs.api")
+			map.n["<leader>R"] = { desc = "Remote" }
+			map.n["<leader>Rc"] = { api.connect, desc = "Connect" }
+			map.n["<leader>Rd"] = { api.disconnect, desc = "Disconnect" }
+			map.n["<leader>Re"] = { api.edit, desc = "Edit" }
+			-- Override telescope find_files and live_grep to make dynamic based on if connected to host
+			local connections = require("remote-sshfs.connections")
+			local builtin = require("telescope.builtin")
+			map.n["<leader>ff"] = {
+				function()
+					if connections.is_connected then
+						api.find_files()
+					else
+						builtin.find_files()
+					end
+				end,
+				desc = "Find files",
+			}
+			map.n["<leader>fw"] = {
+				function()
+					if connections.is_connected then
+						api.live_grep()
+					else
+						builtin.live_grep()
+					end
+				end,
+				desc = "Find files",
+			}
+		end,
```

Comment:
```suggestion
    opts = function(_, opts)
      local map = opts.mappings
      map.n["<Leader>R"] = { desc = "Remote" }
      map.n["<Leader>Rc"] = { function() require("remote-sshfs.api").connect() end, desc = "Connect" }
      map.n["<Leader>Rd"] = { function() require("remote-sshfs.api").disconnect() end, desc = "Disconnect" }
      map.n["<Leader>Re"] = { function() require("remote-sshfs.api").edit() end, desc = "Edit" }
      -- Override telescope find_files and live_grep to make dynamic based on if connected to host
      local find_files = map.n["<Leader>ff"]
      if type(find_files) == "table" then
        local orig_find_files = find_files[1]
        find_files[1] = function()
          if require("remote-sshfs.connections").is_connected then
            require("remote-sshfs.api").find_files()
          elseif type(orig_find_files) == "function" then
            orig_find_files()
          elseif type(orig_find_files) == "string" then
            vim.cmd(orig_find_files)
          end
        end
      end
      local live_grep = map.n["<Leader>fw"]
      if type(live_grep) == "table" then
        local orig_live_grep = live_grep[1]
        live_grep[1] = function()
          if require("remote-sshfs.connections").is_connected then
            require("remote-sshfs.api").live_grep()
          elseif type(orig_live_grep) == "function" then
            orig_live_grep()
          elseif type(orig_live_grep) == "string" then
            vim.cmd(orig_live_grep)
          end
        end
      end
    end,
```

Fixes bugs in lazy loading and improve robustness of overriding original mappings